ZenPacks/zenoss/Microsoft/Windows/handlers.py

```diff
@@ -1,25 +1,33 @@
 ##############################################################################
 #
-# Copyright (C) Zenoss, Inc. 2013, all rights reserved.
+# Copyright (C) Zenoss, Inc. 2016, 2017, all rights reserved.
 #
 # This content is made available according to terms specified in
 # License.zenoss under the directory where your Zenoss product is installed.
 #
 ##############################################################################
 
+import logging
 from zope.component import adapter
 from OFS.interfaces import IObjectWillBeMovedEvent
 from OFS.event import ObjectWillBeRemovedEvent
 
 from datasources.ServiceDataSource import ServiceDataSource
 from jobs import ReindexWinServices
 
+log = logging.getLogger('zen.MicrosoftWindows')
 DEFAULT_SERVICE = '/zport/dmd/Devices/Server/Microsoft/rrdTemplates/WinService/datasources/DefaultService'
 
 
 @adapter(ServiceDataSource, IObjectWillBeMovedEvent)
 def onServiceDataSourceMoved(ob, event):
     if isinstance(event, ObjectWillBeRemovedEvent):
         dmd = ob.getDmdRoot("Devices")
-        if dmd:
+        try:
+            template = ob.getPrimaryUrlPath().split('/')[-3]
```

Comment:
could this be done as ob.rrdTemplate().id, or something like that?  Might be a little bit clearer?